python/custreamz/custreamz/kafka.py

```diff
@@ -0,0 +1,156 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+import confluent_kafka as ck
+from cudf_kafka._lib.kafka import KafkaDatasource
+
+import cudf
+
+from custreamz.utils import docutils
+
+
+# Base class for anything class that needs to interact with Apache Kafka
+class CudfKafkaClient(object):
+    def __init__(self, kafka_configs):
+        self.kafka_configs = kafka_configs
+
+        self.kafka_confs = {}
+        for key, value in self.kafka_configs.items():
+            self.kafka_confs[str.encode(key)] = str.encode(value)
+
+        self.kafka_meta_client = KafkaDatasource(self.kafka_confs)
+
+    @docutils.doc_unsubscribe()
+    def unsubscribe(self):
+
+        """{docstring}"""
+
+        return self.kafka_meta_client.unsubscribe()
+
+    @docutils.doc_close()
+    def close(self, timeout=10000):
+
+        """{docstring}"""
+
+        return self.kafka_meta_client.close(timeout=timeout)
+
+
+# Apache Kafka Consumer implementation
+class Consumer(CudfKafkaClient):
+    def __init__(self, kafka_configs):
+        super().__init__(kafka_configs)
+
+    @docutils.doc_read_gdf()
+    def read_gdf(
+        self,
+        topic=None,
+        partition=0,
+        lines=True,
+        start=0,
+        end=0,
+        batch_timeout=10000,
+        delimiter="\n",
+        message_format="json",
+    ):
+
+        """{docstring}"""
+
+        if topic is None:
+            raise ValueError(
+                "ERROR: You must specifiy the topic "
+                + "that you want to consume from"
+            )
+
+        kafka_datasource = KafkaDatasource(
+            self.kafka_confs,
+            topic.encode(),
+            partition,
+            start,
+            end,
+            batch_timeout,
+            delimiter.encode(),
+        )
+
+        if message_format.lower() == "json":
+            result = cudf.io.read_json(self.kafka_datasource)
+        elif message_format.lower() == "csv":
+            result = cudf.io.read_csv(self.kafka_datasource)
+        elif message_format.lower() == "orc":
+            result = cudf.io.read_orc(self.kafka_datasource)
+        elif message_format.lower() == "avro":
+            result = cudf.io.read_avro(self.kafka_datasource)
+        elif message_format.lower() == "parquet":
+            result = cudf.io.read_parquet(self.kafka_datasource)
```

Comment:
Maybe have a dictionary that looks like: `{"json": cudf.io.read_json, "csv": cudf.io.read_csv, ...}` and use that instead of these `if` statements